### ISSUE TITLE
Pagination fails for GitLab APIs with more than 10000 items

### ISSUE THREAD

**Issue by lorenzo-cavazzi** [SwissDataScienceCenter/renku-ui#2055]
## Description

GitLab "traditional" pagination (i.e. offset-based pagination) partially fails when returning more than 10'000 items ([docs reference](https://docs.gitlab.com/ee/user/gitlab_com/index.html#pagination-response-headers)). In that case, some headers are missing, among which `x-total` and `x-total-pages` ([docs reference](https://docs.gitlab.com/ee/api/index.html#pagination-response-headers)).
Since we use those fields for our pagination system, it currently fails on >10'000 items

## Reproduce

Admins on renkulab.io already get this error when listing all projects https://renkulab.io/projects/all

All the users will soon have this problem since we already have  >9'500 public projects.

![image](https://user-images.githubusercontent.com/43481553/193576808-d9f26d28-b942-463c-96d2-4192073d4b99.png)

## Possible solutions

I haven't checked it thoroughly yet, but here are 2 possible solutions:
- Manually adding the missing fields/data in those cases where they are missing and we still get items (e.g. if we get items but no `x-total`, we give for granted there are more than 10'000 items). In that case, we can still show the pages up to `10000 / <single_page_size>` and specify on the last page that some results are missing and a more specific search is required.
- Try the keyset pagination ([docs reference](https://docs.gitlab.com/ee/api/#keyset-based-pagination)).

I would go for the quickest option since this might be solved in a better way with our KG-based search.

**Comment by ciyer** [comment]
This problem should go away when we switch to the KG for search, right?

**Comment by lorenzo-cavazzi** [comment]
Yes, unless we use the projects API for anything else (I'm not 100% up to date with all the changes in the kg-search branch)

**Comment by lorenzo-cavazzi** [comment]
[Renku 2.0](https://blog.renkulab.io/renku-2) will disentangle Projects and Repositories, so we won't rely on GitLab APIs that much anymore
